exp/api/v1alpha3/azuremachinepool_types.go

```diff
@@ -0,0 +1,148 @@
+		ProviderIDList []string `json:"providerIDList,omitempty"`
+	}
+
+	// AzureMachinePoolStatus defines the observed state of AzureMachinePool
+	AzureMachinePoolStatus struct {
+		// Ready is true when the provider resource is ready.
+		// +optional
+		Ready bool `json:"ready"`
+
+		// Replicas is the most recently observed number of replicas.
+		// +optional
+		Replicas int32 `json:"replicas"`
+
+		// VMState is the provisioning state of the Azure virtual machine.
+		// +optional
+		ProvisioningState *infrav1.VMState `json:"provisioningState,omitempty"`
+
+		// ErrorReason will be set in the event that there is a terminal problem
+		// reconciling the MachinePool and will contain a succinct value suitable
+		// for machine interpretation.
+		//
+		// This field should not be set for transitive errors that a controller
+		// faces that are expected to be fixed automatically over
+		// time (like service outages), but instead indicate that something is
+		// fundamentally wrong with the MachinePool's spec or the configuration of
+		// the controller, and that manual intervention is required. Examples
+		// of terminal errors would be invalid combinations of settings in the
+		// spec, values that are unsupported by the controller, or the
+		// responsible controller itself being critically misconfigured.
+		//
+		// Any transient errors that occur during the reconciliation of MachinePools
+		// can be added as events to the MachinePool object and/or logged in the
+		// controller's output.
+		// +optional
+		FailureReason *errors.MachineStatusError `json:"failureReason,omitempty"`
+
+		// ErrorMessage will be set in the event that there is a terminal problem
+		// reconciling the MachinePool and will contain a more verbose string suitable
+		// for logging and human consumption.
+		//
+		// This field should not be set for transitive errors that a controller
+		// faces that are expected to be fixed automatically over
+		// time (like service outages), but instead indicate that something is
+		// fundamentally wrong with the MachinePool's spec or the configuration of
+		// the controller, and that manual intervention is required. Examples
+		// of terminal errors would be invalid combinations of settings in the
+		// spec, values that are unsupported by the controller, or the
+		// responsible controller itself being critically misconfigured.
+		//
+		// Any transient errors that occur during the reconciliation of MachinePools
+		// can be added as events to the MachinePool object and/or logged in the
+		// controller's output.
+		// +optional
+		FailureMessage *string `json:"failureMessage,omitempty"`
+	}
+
+	// +kubebuilder:object:root=true
+	// +kubebuilder:subresource:status
+	// +kubebuilder:printcolumn:name="Replicas",type="string",JSONPath=".status.replicas",description="AzureMachinePool replicas count"
+	// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="AzureMachinePool replicas count"
```

Comment:
Should we add the same lower priority printcolums as AzureMachine to print Cluster, MachinePool, VMSize and VMSS ID in `-o wide` mode?

Reply:
Looks great!